etl/steps/__init__.py

```diff
@@ -98,6 +88,50 @@ def to_dependency_order(
     return filtered
 
 
+def filter_to_subgraph(
+    graph: Graph, includes: Iterable[str], incl_forward: bool = True
+) -> Graph:
+    """
+    Filter the full graph to only the included nodes, and all their dependencies.
+
+    If the incl_forward flag is true, also include "forward dependencies" (dependents),
+    and their own (backward) dependencies.
+
+    Assumes that the graph is organized dependent -> dependency (A -> B means A is
+    dependent on B).
+    """
+    all_steps = graph_nodes(graph)
+    included = {
+        s for s in all_steps if any(re.findall(pattern, s) for pattern in includes)
+    }
+
+    if incl_forward:
+        # Reverse the graph to find all nodes dependent on included nodes (forward deps)
+        forward_deps = set(traverse(reverse_graph(graph), included))
+        included = included.union(forward_deps)
+
+    # Now traverse the other way to find all dependencies of included nodes (backward deps)
+    return traverse(graph, included)
+
+
+def traverse(graph: Graph, nodes: Set[str]) -> Graph:
```

Comment:
The terminology `traverse` is too neutral for a DAG, where you are only traversing in one direction.

If the graph is in `node -> dependencies` direction, then this would be better called `ancestor_subgraph()`

Reply:
👍  you're right that it's not a full traversal - but also it's used in both logical directions if you look in the above code, and is meant to be general in that way (to avoid rewriting the same logic in the other direction like we were doing before). 

Maybe just `bfs` or `find_reachable` then?